src/util/pagination.js

```diff
@@ -14,8 +14,46 @@ function getPaginationMeta ({ nbResults, page, nbResultsPerPage }) {
   return paginationMeta
 }
 
+async function offsetPaginate ({
+  queryBuilder,
+  orderBy,
+  order,
+  nbResultsPerPage,
+  page,
+  applyOrder = true,
+}) {
+  // Clone the query builder to have the count for all matched results before pagination filtering
+  const countQueryBuilder = queryBuilder.clone()
+
+  if (applyOrder) {
+    queryBuilder.orderBy(orderBy, order)
```

Comment:
There is only one case where the order params must not be applied:
https://github.com/stelace/stelace/blob/0807210e6b8e4138c376586cdd667617d8ae4815/src/services/document.js#L330

This is because the document stats aggregation already performs it with the WITH CLAUSE queries.
I should have added a comment mentioning the use at that exact location.

Reply:
This is not what I meant: why add the `applyOrder` parameter when you can just see if `orderBy` and `order` are provided?

A parameter would only make sense if you wanted to compute it dynamically (when you don’t know in advance you have to provide `orderBy` and `order` just in case).

Reply:
Ok I misunderstood. No need to introduce this `applyOrder` then. Thanks!